File: php/features.php
<?php
declare(strict_types=1);

// Sharedmobilitych SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SharedmobilitychFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SharedmobilitychBaseFeature();
            case "test":
                return new SharedmobilitychTestFeature();
            default:
                return new SharedmobilitychBaseFeature();
        }
    }
}

File: py/features.py
# Sharedmobilitych SDK feature factory

from feature.base_feature import SharedmobilitychBaseFeature
from feature.test_feature import SharedmobilitychTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SharedmobilitychBaseFeature(),
        "test": lambda: SharedmobilitychTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
